src/Math-Matrix/PMNDArray.class.st

```diff
@@ -97,6 +97,21 @@ PMNDArray >> fromNestedArray: aFlatArray withShape: aShape [
 	shape ifNotEmpty: [ self updateStrides]
 ]
 
+{ #category : #operation }
+PMNDArray >> hadamardProduct: aPMNDArray [
+	(self hasSameShapeAs: aPMNDArray)
+	ifTrue:[
```

Comment:
This is the normal flow, is that correct? If so, consider moving this out of the block and have run after the check. `ifFalse` becomes a guard clause. The code, in my opinion, then becomes a little more maintainable.

Reply:
Agree, something like:
```
(self hasSameShapeAs:aPMNDArray)
ifFalse:[ShapeMismatch signal].
^ self class new array: ((self asArray) with: (aPMNDArray asArray) collect:[:a :b| a*b])
		withShape: self shape.
```

```